### PR TITLE
Add Translate V3 client

### PR DESCRIPTION
**Pull request by blowmage**

### CODE REVIEW

File: google-cloud-translate/lib/google-cloud-translate.rb [hunk 1]
```diff
@@ -131,14 +116,10 @@ def translate key = nil, scope: nil, retries: nil, timeout: nil
...
-    def self.translate key = nil, project_id: nil, credentials: nil, scope: nil,
-                       retries: nil, timeout: nil, project: nil, keyfile: nil
-      require "google/cloud/translate"
-      Google::Cloud::Translate.new key: key, project_id: project_id,
-                                   credentials: credentials,
-                                   scope: scope, retries: retries,
-                                   timeout: timeout,
-                                   project: project, keyfile: keyfile
+    def self.translate key = nil, project_id: nil, credentials: nil, scope: nil, retries: nil, timeout: nil
+      require "google/cloud/translate/v2"
```
Comment (dazuma) [inline google-cloud-translate/lib/google-cloud-translate.rb]: These should probably also default to V3 (and have a `:version` argument to select a version).

Comment (blowmage) [reply google-cloud-translate/lib/google-cloud-translate.rb]: I'm waiting to get all the samples working before taking another pass at the inline docs and the guides. My guess is we will show the V3 code examples before the V2 code examples.

Comment (blowmage) [reply google-cloud-translate/lib/google-cloud-translate.rb]: This has been updated with a v3 client example, but it isn't using `version: :v3`. PTAL.

Comment (blowmage) [reply google-cloud-translate/lib/google-cloud-translate.rb]: The v3 client doesn't support API Keys, only the legacy v2 client supports them.

File: google-cloud-translate/OVERVIEW.md [hunk 2]
```diff
@@ -22,18 +26,219 @@ key and the project and OAuth 2.0 credentials are provided, the API key
...
+```
+
+## Using the legacy v2 client
+
+### Translating texts
 
 Translating text from one language to another is easy (and extremely
 fast.) The only required arguments to
-{Google::Cloud::Translate::Api#translate} are a string and the [ISO
+{Google::Cloud::Translate::::Api#translate} are a string and the [ISO
```
Comment (dazuma) [inline google-cloud-translate/OVERVIEW.md]: whoops `::::`

**Comment by dazuma** [comment]
Are we going to update `Google::Cloud.translate` and `Google::Cloud#translate`?

**Comment by blowmage** [comment]
> Are we going to update `Google::Cloud.translate` and `Google::Cloud#translate`?

I wasn't planning on it, but it looks like they should those default to the v2 client, yes?

**Comment by dazuma** [comment]
I think everything should default to the v3 client. (Anyway it would be confusing if some of the "global" methods defaulted to v2 and some to v3.)

**Comment by blowmage** [comment]
Updated to default the `Google::Cloud` methods to the v3 client. (I had to skip one of the tests because it would not clear the stub and was causing the generated v3 tests to fail.)

File: google-cloud-translate/.rubocop.yml [hunk 3]
```diff
@@ -4,12 +4,22 @@ inherit_gem:
 AllCops:
   Exclude:
     - "acceptance/**/*"
-    - "google-cloud-translate.gemspec"
-    - "Rakefile"
+    # - "google-cloud-translate.gemspec"
+    - "lib/google/cloud/translate/v3.rb"
+    - "lib/google/cloud/translate/v3/**/*"
+    # - "Rakefile"
```
Comment (dazuma) [inline google-cloud-translate/.rubocop.yml]: Drop these commented out lines if not needed.

Comment (blowmage) [reply google-cloud-translate/.rubocop.yml]: Oops.

**Pull request merged**